File: INVBINCF.py
import math
def comb(n,k):
    f=math.factorial
    com=f(n)/(f(k)*f(n-k))
    return int(com)
test_cases=int(input())
for k in range(0,test_cases):
    x=input()
    n,R=x.split()
    n=int(n)
    R=int(R)
    p=1
    div=pow(2,n)
    up=div-1
    for k in range(0,up+1):
        c=comb(up,k)
        if((c%div)==R):
                p=0
                ans=k
                break
    if(p==0):
        print(ans)
    else:
        print(-1)

File: INVBINCF.c
#include<stdio.h>

main()
{
    int i,test_cases,k,p,ans,n,R,up;
    scanf("%d",&test_cases);

    for(k=0;k<test_cases;k++)
    {
        scanf("%d %d",&n,&R);
        up=pow(2,n-1);
        for(k=0;k<=up;k++)
        {
            if((comb(up,k)%R)==0)
            {
                p=0;
                ans=k;
                break;
            }
        }

        if(p==0)
            printf("%d\n",ans);
        else
                printf("-1\n");

    }

 return 0;
}
